File: app/src/main/java/com/kota/Bahamut/Dialogs/DialogShortenUrlItemViewAdapter.java
package com.kota.Bahamut.Dialogs;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.kota.Bahamut.DataModels.ShortenUrl;
import com.kota.Bahamut.R;

import java.util.List;

public class DialogShortenUrlItemViewAdapter extends RecyclerView.Adapter<DialogShortenUrlViewHolder> {
    private final List<ShortenUrl> _blocklist;
    private DialogShortenUrlItemViewListener mClickListener;

    public DialogShortenUrlItemViewAdapter(List<ShortenUrl> dataSet) {
        _blocklist = dataSet;
    }

    @NonNull
    @Override
    public DialogShortenUrlViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.shorten_url_item_view, parent, false);
        return new DialogShortenUrlViewHolder(v, mClickListener);
    }

    @Override
    public void onBindViewHolder(@NonNull DialogShortenUrlViewHolder holder, int position) {
        ShortenUrl obj = getItem(position);
        holder.setTitle(obj.getTitle());
        holder.setDescription(obj.getDescription());
    }

    public ShortenUrl getItem(int position) {
        return this._blocklist.get(position);
    }

    @Override
    public int getItemCount() {
        return _blocklist.size();
    }

    public void setOnItemClickListener(DialogShortenUrlItemViewListener listener) {
        this.mClickListener = listener;
    }
}

File: app/src/main/java/com/kota/Bahamut/Dialogs/DialogShortenUrlItemViewAdapter.kt
package com.kota.Bahamut.Dialogs

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.kota.Bahamut.DataModels.ShortenUrl
import com.kota.Bahamut.R

class DialogShortenUrlItemViewAdapter(private val _blocklist: List<ShortenUrl>) : 
    RecyclerView.Adapter<DialogShortenUrlViewHolder>() {
    
    private var mClickListener: DialogShortenUrlItemViewListener? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DialogShortenUrlViewHolder {
        val v = LayoutInflater.from(parent.context)
            .inflate(R.layout.shorten_url_item_view, parent, false)
        return DialogShortenUrlViewHolder(v, mClickListener)
    }

    override fun onBindViewHolder(holder: DialogShortenUrlViewHolder, position: Int) {
        val obj = getItem(position)
        holder.setTitle(obj.title)
        holder.setDescription(obj.description)
    }

    fun getItem(position: Int): ShortenUrl {
        return _blocklist[position]
    }

    override fun getItemCount(): Int {
        return _blocklist.size
    }

    fun setOnItemClickListener(listener: DialogShortenUrlItemViewListener?) {
        mClickListener = listener
    }
}
